src/conntrack_store.c

```diff
@@ -194,22 +194,22 @@ handle_update_event(struct conntrack_store *conn_store,
     const gpointer key = GUINT_TO_POINTER(ct_id);
     pthread_mutex_lock(&conn_store->lock);
     ct_entry = g_hash_table_lookup(conn_store->store, key);
-    pthread_mutex_unlock(&conn_store->lock);
     if (ct_entry == NULL) {
         LOG(VERBOSE, "%s: Update received for a non-existent entry. "
             "Treating it as NEW.", __func__);
-        handle_new_event(conn_store, ct);
-        return;
+        res_ct_entry = conntrack_entry_from_nf_conntrack(ct);
+    } else {
+        res_ct_entry = get_conntrack_entry_from_update(ct_entry, ct);
     }
 
-    res_ct_entry = get_conntrack_entry_from_update(ct_entry, ct);
     if (res_ct_entry == NULL) {
         LOG(WARNING, "%s: received res_ct_entry as NULL", __func__);
-        return;
+        goto finish;
     }
 
-    pthread_mutex_lock(&conn_store->lock);
-    g_hash_table_replace(conn_store->store, key, res_ct_entry);
+    g_hash_table_insert(conn_store->store, key, res_ct_entry);
```

Comment:
tbh no reason.
when going thru the docs i found they were doing the same thing. So i used`g_hash_table_insert` instead of `g_hash_table_replace` as we were using insert for the NEW event case as well.
lmk if this is fine otherwise i can revert it back to `g_hash_table_replace` as well.